src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1283,7 +1296,7 @@ bool isStructOrRecordKeyword(SyntaxToken token)
 
         private bool ShouldAsyncBeTreatedAsModifier(bool parsingStatementNotDeclaration)
         {
-            Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.AsyncKeyword);
+            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IdentifierToken && GetModifier(this.CurrentToken) != DeclarationModifiers.None);
```

Comment:
>Debug.Assert [](http://example.com/codeflow?start=12&length=12)

The updated assert is from [https://github.com/dotnet/roslyn/pull/60823](https://github.com/dotnet/roslyn/pull/60823) where this method is also renamed.